--- R385-v1/PrePasses.ml
+++ R385-v2/PrePasses.ml
@@ -994,8 +994,8 @@
         { crate with
           type_decls = TypeDeclId.Map.add id decl crate.type_decls;
           declarations = TypeGroup (NonRecGroup id) :: crate.declarations;
-          item_names = (TypeId id, name) :: crate.item_names;
-          short_names = (TypeId id, name) :: crate.short_names }
+          item_names = (IdType id, name) :: crate.item_names;
+          short_names = (IdType id, name) :: crate.short_names }
       in
       (crate, Some id)
 
